plugins/discovery/endpoint_creator.go

```diff
@@ -0,0 +1,72 @@
+package discovery
+
+import (
+	"github.com/sirupsen/logrus"
+	"github.com/wavefronthq/wavefront-collector-for-kubernetes/internal/discovery"
+)
+
+type endpointCreator struct {
+	delegates                  map[string]*delegate
+	providers                  map[string]discovery.ProviderInfo
+	disableAnnotationDiscovery bool
+}
+
+func (e *endpointCreator) discoverEndpointsWithRules(resource discovery.Resource) []*discovery.Endpoint {
+	var eps []*discovery.Endpoint
+	for _, delegate := range e.delegates {
+		if delegate.filter.matches(resource) {
+			if ep := e.makeEndpoint(resource, delegate.plugin); ep != nil {
+				eps = append(eps, ep)
+			}
+		}
+	}
+	return eps
+}
+
+func (e *endpointCreator) discoverEndpointsWithAnnotations(resource discovery.Resource) []*discovery.Endpoint {
+	var eps []*discovery.Endpoint
+	// delegate to runtime handlers if no matching delegate
+	if ep := e.makeEndpoint(resource, discovery.PluginConfig{Type: "prometheus", Name: "example"}); ep != nil {
```

Comment:
Should probably remove the `example` name since that is used to determine whether to check for annotations here: https://github.com/wavefrontHQ/wavefront-collector-for-kubernetes/blob/b12b7bb65adb283811abadcb6d88e564ea21bbc2/plugins/discovery/prometheus/encoder.go#L122